wagtail_transfer/views.py

```diff
@@ -86,10 +133,31 @@ def choose_page(request):
 
 @require_POST
 def do_import(request):
-    response = requests.get(f"http://localhost:8000/wagtail-transfer/api/pages/{request.POST['source_page_id']}/")
+    source_config = settings.WAGTAILTRANSFER_SOURCES[request.POST['source']]
+    response = requests.get(f"{source_config['BASE_URL']}api/pages/{request.POST['source_page_id']}/")
 
     importer = ImportPlanner(request.POST['source_page_id'], request.POST['dest_page_id'])
     importer.add_json(response.content)
+
+    while importer.missing_object_data:
+        # convert missing_object_data from a set of (model_class, id) tuples
+        # into a dict of {model_class_label: [list_of_ids]}
+        missing_object_data_by_type = defaultdict(list)
+        for model_class, source_id in importer.missing_object_data:
+            missing_object_data_by_type[model_class].append(source_id)
+
+        request_data = {
+            model_class._meta.label_lower: ids
+            for model_class, ids in missing_object_data_by_type.items()
+        }
+
+        # request the missing object data and add to the import plan
+        response = requests.post(
+            f"{source_config['BASE_URL']}api/objects/",
+            json=request_data
+        )
+        importer.add_json(response.content)
```

Comment:
If one of the IDs are missing on the source, this would probably loop indefinitely as the source doesn't return an error.

Maybe we should check if there are any objects in the response and quit if not?

Reply:
As discussed, I've bumped this to a separate ticket - https://github.com/wagtail/wagtail-transfer/issues/8